src/lib/converter.js

```diff
@@ -1,94 +1,138 @@
-        // -q : Supress log output except on error
-        // --follow : Follow symlinked directories
-        // -r : Recursively watch directories or files
-        // --watch : Watch a directory or file
-        var nodeArgs = [sassPath, appDir, '--output', appDir, '--output-style', 'compressed', '-q', '--follow', '--importer', importerPath];
-        if (options.watch) {
-            nodeArgs.push('-r', '--watch');
-        }
-
-        logger.trace(process.execPath, nodeArgs.join(' '));
-        var env = Object.create( process.env );
-        env.PROJECT_DIR = projectDir;
-        env.APP_DIR = appDir;
-        currentSassProcess = spawn(process.execPath, nodeArgs, { env: env });
-
-        var isResolved = false;
-        var watchResolveTimeout;
-        currentSassProcess.stdout.on('data', function (data) {
-            var stringData = data.toString();
-            logger.info(stringData);
-        });
-
-        currentSassProcess.stderr.on('data', function (err) {
-            var message = '';
-            var stringData = err.toString();
-
-            try {
-                var parsed = JSON.parse(stringData);
-                message = parsed.formatted || parsed.message || stringData;
-            } catch (e) {
-                renderMsg = true;
-                message = err.toString();
-            }
-
-            logger.info(message);
-        });
-
-        currentSassProcess.on('error', function (err) {
-            logger.info(err.message);
-            if (!isResolved) {
-                isResolved = true;
-                reject(err);
-            }
-        });
-
-        // TODO: Consider using close event instead of exit
-        currentSassProcess.on('exit', function (code, signal) {
-            currentSassProcess = null;
-            if (!isResolved) {
-                isResolved = true;
-                if (code === 0) {
-                    resolve();
-                } else {
-                    reject(Error('SASS compiler failed with exit code ' + code));
-                }
-            }
-        });
-
-        // SASS does not recompile on watch, so directly resolve.
-        if (options.watch && !isResolved) {
-            isResolved = true;
-            resolve();
-        }
-    });
+	options = options || {};
+	var sassPath = getSassPath();
+	var data = {
+		sassPath: sassPath,
+		projectDir: projectDir,
+		appDir: appDir,
+		logger: logger,
+		options: options
+	};
+	
+	if (options.watch) {
+		createWatcher(data);
+	}
+
+	return spawnNodeSass(data);
+}
+
+function getWatcher() {
+	return watcher;
+}
+
+function createWatcher(data) {
+	var appDir = data.appDir;
+	var watcherOptions = {
+		ignoreInitial: true,
+		cwd: appDir,
+		awaitWriteFinish: {
+			pollInterval: 100,
+			stabilityThreshold: 500
+		},
+		ignored: ['**/.*', '.*'] // hidden files
+	};
+	
+	watcher = choki.watch(['**/*.scss', '**/*.sass'], watcherOptions)
+		.on('all', (event, filePath) => {
+			spawnNodeSass(data);
```

Comment:
In case you apply several changes and save them one-by-one, you'll receive multiple events and several processes will be spawned. This is a problem, so I suggest chaining the calls to `spawnNodeSass`:
```JavaScript
var watchPromisesChain = Promise.resolve();
watcher = choki.watch(['**/*.scss', '**/*.sass'], watcherOptions)
	.on('all', (event, filePath) => {
		watchPromiseChain = watchPromiseChain.then( () => spawnNodeSass(data) );
	});
```